### PR TITLE
Fixing team/ROA to support admin interface

### PR DESCRIPTION
**Pull request by andrewhayward**
I've made a few changes to the `Team` model and the `ROA` infrastructure, which allows us to use the standard Django admin site to make changes to `ROA` models.

### CODE REVIEW

**Comment by andrewhayward** [comment]
@christensenep - thoughts?

File: mozbadges/site/teams/models.py [hunk 1]
```diff
@@ -5,12 +5,11 @@
 from rest_framework import serializers
 
 class Team (ROAModel):
-    id = models.IntegerField()
     name = models.CharField(max_length=255)
     slug = models.CharField(max_length=255, primary_key=True)
     url = models.URLField()
     description = models.TextField(blank=True, null=True)
-    email = models.EmailField(blank=True, null=True)
+    email = models.EmailField()
```
Comment (andrewhayward) [inline mozbadges/site/teams/models.py:12]: `email` is seemingly required by the API.

File: mozbadges/site/teams/models.py [hunk 2]
```diff
@@ -5,12 +5,11 @@
 from rest_framework import serializers
 
 class Team (ROAModel):
-    id = models.IntegerField()
```
Comment (andrewhayward) [inline mozbadges/site/teams/models.py:8]: I really don't know why we ever decided to surface this field in the API, but regardless, we don't need it, and getting rid of it makes things a lot easier.

File: vendor-local/src/django-roa/django_roa/db/models.py [hunk 3]
```diff
@@ -529,7 +530,7 @@ def delete(self):
         logger.debug(u"""Deleting  : "%s" through %s""" % \
             (unicode(self), unicode(resource.uri)))
 
-        result = resource.delete(**ROA_CUSTOM_ARGS)
+        result = resource.delete(headers={'Content-Type': 'application/json'}, **ROA_CUSTOM_ARGS)
```
Comment (andrewhayward) [inline vendor-local/src/django-roa/django_roa/db/models.py:533]: The API doesn't like the default `application/octet-stream` content type, so we're having to force it here.

File: mozbadges/site/teams/models.py [hunk 4]
```diff
@@ -23,6 +22,10 @@ def serializer():
     def get_resource_url_list (): 
         return urljoin(settings.BADGEKIT_API_ENDPOINT, '/systems/' + settings.BADGEKIT_API_SYSTEM + '/issuers')
 
+    @staticmethod
+    def get_resource_url_count ():
+        return Team.get_resource_url_list()
```
Comment (andrewhayward) [inline mozbadges/site/teams/models.py:27]: We need to override this to get the right URL.

File: mozbadges/site/teams/models.py (hunk 2 above)
Comment (ghost) [inline mozbadges/site/teams/models.py:8]: Yeah, I sort of blindly included this, but it certainly is an odd thing to expose.  No objections here.

File: mozbadges/site/teams/models.py (hunk 1 above)
Comment (ghost) [inline mozbadges/site/teams/models.py:12]: Oops, good catch.

**Comment by ghost** [comment]
Looks good to me.

**Pull request merged**